### PR TITLE
Create devcontainer.json

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
See https://code.visualstudio.com/docs/remote/devcontainerjson-reference for more information.

### CODE REVIEW

File: .github/workflows/test-deploy.yml [hunk 1]
```diff
@@ -22,10 +22,7 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: ${{ github.event.pull_request.head.sha || github.ref }}
-      - uses: actions/setup-node@v2
-      - name: check lock
-        run: |
-          test $(cat package-lock.json | jq .lockfileVersion) == "2"
+      - run: jq --exit-status .lockfileVersion==2 < package-lock.json
```
Comment (dacbd) [inline .github/workflows/test-deploy.yml:25]: can we name it for verboseness?

File: .devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "image": "mcr.microsoft.com/vscode/devcontainers/javascript-node:16",
+  "hostRequirements": {
+    "cpus": 4,
+    "memory": "8gb",
+    "storage": "32gb"
```
Comment (DavidGOrtega) [inline .devcontainer.json:6]: What will happen if I can not provide such requirements?

**Review by DavidGOrtega — CHANGES_REQUESTED** [review]

File: .github/workflows/test-deploy.yml (hunk 1 above)
Comment (0x2b3bfa0) [reply .github/workflows/test-deploy.yml:25]: I usually avoid names when the command intent is “evident” but sure

File: .devcontainer.json (hunk 2 above)
Comment (0x2b3bfa0) [reply .devcontainer.json:6]: That's for Codespaces, not for your computer. 😅

Comment (0x2b3bfa0) [reply .devcontainer.json:6]: https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_minimum-host-requirements

Comment (0x2b3bfa0) [reply .devcontainer.json:6]: I authorize you to develop on an Intel 80386.

Comment (DavidGOrtega) [reply .devcontainer.json:6]: I can run it still locally right?

File: .github/workflows/test-deploy.yml (hunk 1 above)
Comment (0x2b3bfa0) [inline .github/workflows/test-deploy.yml:25]:
```suggestion
      - name: check that npm lock file is on version 2
        run: jq --exit-status .lockfileVersion==2 < package-lock.json
```

```js
if (check == true) { // if the check variable is true
  return;
}
```

File: .devcontainer.json (hunk 2 above)
Comment (DavidGOrtega) [reply .devcontainer.json:6]: Ok so it just displays a warning...

> I authorize you to develop on an Intel 80386.

If you need such an amount of resources to dev CML...

File: .github/workflows/test-deploy.yml (hunk 1 above)
Comment (dacbd) [reply .github/workflows/test-deploy.yml]: lol well there is no guessing what its for now 🙃 ❤️

Comment (0x2b3bfa0) [reply .github/workflows/test-deploy.yml]: Anyway, pinging @casperdcl for opinions on terse pipeline syntax. (Help!) 😄

Comment (0x2b3bfa0) [reply .github/workflows/test-deploy.yml]: Resolved with https://github.com/iterative/cml/pull/1082#discussion_r910301999

File: .devcontainer.json (hunk 2 above)
Comment (0x2b3bfa0) [reply .devcontainer.json:6]: > _it just displays a warning..._

Yes, nothing else

Comment (0x2b3bfa0) [reply .devcontainer.json:6]: > _If you need such an amount of resources to dev CML..._

Codespaces machines with less than that are too slow. 🐢

File: .github/workflows/test-deploy.yml (hunk 1 above)
Comment (dacbd) [reply .github/workflows/test-deploy.yml]: I take it back the action's job name is sufficient for my `at a glance` determination of possible reasons for a check failure which is why names are nice

File: .devcontainer.json (hunk 2 above)
Comment (DavidGOrtega) [reply .devcontainer.json:6]: I do not pretend to use Codespaces my connection sometimes is not great

File: .github/workflows/test-deploy.yml (hunk 1 above)
Comment (0x2b3bfa0) [reply .github/workflows/test-deploy.yml]: I used to write all my steps with names until I joined Iterative. 😅 In sentence case, for consistency with the default step names.

File: .devcontainer.json (hunk 2 above)
Comment (0x2b3bfa0) [reply .devcontainer.json:6]: But I do 🙃

**Review by casperdcl — COMMENTED** [review]
nitpicks:

- maybe add DVC VSCode extension?
- automatically do local npm install the checked out CML repo?
- can the config file live under `./github/` to avoid root clutter?

**Comment by 0x2b3bfa0** [comment]
> _maybe add DVC VSCode extension?_

Makes more sense in the extension repository than in this one, to my mind. Once we have better (i.e. any) integration scenarios and we need it during development, it will be an awesome addition, though.

> _automatically do local npm install (after clone)?_

* https://github.com/iterative/cml/pull/1085

> _can the config file live under ./github/ to avoid root clutter?_

* https://github.com/microsoft/vscode-remote-release/issues/2413

Moreover, it's not specific to GitHub Codespaces, but rather a Visual Studio Code feature.

**Pull request merged**